quinn-proto/src/connection/streams/mod.rs

```diff
@@ -237,7 +245,7 @@ impl<'a> SendStream<'a> {
 
     /// Check if this stream was stopped, get the reason if it was
     pub fn stopped(&mut self) -> Result<Option<VarInt>, UnknownStream> {
-        match self.state.send.get(&self.id) {
+        match self.state.send.get(&self.id).and_then(|s| s.as_ref()) {
```

Comment:
This is incorrect. We should return `Ok(None)` if the stream is unallocated. Otherwise, it becomes impossible to distinguish open but unallocated send streams from closed and forgotten ones.

Reply:
A deallocated stream should return `None` from `self.state.send.get`, I'm not deallocating the inner `Option` ever. Maybe I'm misunderstanding though.

Reply:
By "unallocated stream" I mean a stream that exists in `send`, but is associated with a value of `None`, so `get` here returns `Some(&mut None)`. Such streams might be surfaced to the user e.g. after receiving a new remotely-initiated bidirectional stream.